### PR TITLE
Ensure Clang/GCC are at least the minimum version

### LINKED ISSUE #960: binary distro error on Ubuntu 14.04 (version `GLIBCXX_3.4.18' not found ).  The symlink trick for g++-4.4 didn't work for this user.
I get the following error when I try to execute runMixedIntegerForest.m. I have istalled your latest binary on Ubuntu 14.04. My gcc and g++ versions are 4.9.  

Error message:

Invalid MEX-file
'/home/avinash/drake-distro/drake/systems/plants/constructModelmex.mexa64':
/usr/local/MATLAB/R2014b/bin/glnxa64/../../sys/os/glnxa64/libstdc++.so.6:
version `GLIBCXX_3.4.18' not found (required by
/home/avinash/drake-distro/drake/pod-build/lib/libdrakeCollision.so)

Error in RigidBodyManipulator/createMexPointer (line 2093)
        obj.mex_model_ptr = constructModelmex(obj);

Error in RigidBodyManipulator/compile (line 831)
      model = createMexPointer(model);

Error in RigidBodyManipulator/addRobotFromURDF (line 59)
model = compile(model);  % ideally this would happen on entry into any
function...

Error in RigidBodyManipulator (line 73)
          obj =
          addRobotFromURDF(obj,filename,zeros(3,1),zeros(3,1),options);

Error in Quadrotor (line 11)
      obj =
      obj@RigidBodyManipulator(getFullPathFromRelativePath('quadrotor.urdf'),options);
      
Error in runMixedIntegerForest (line 5)
r = Quadrotor();

### LINKED ISSUE #1281: install_prereqs has to be run twice (before and after download-all)
hi guys,

just wanted to let you know that for my installation i encountered problems with ccmake (on the 'make options' command) and glut.h (on a 'make' command). So i was missing
```
$ sudo apt-get install cmake-curses-gui freeglut3-dev
```
to make it work.

Best regards from Munich,
Gerold

### LINKED ISSUE #1649: Building master results in link errors on g++ 4.8
On 4.8.5:
```
Linking CXX executable ../../bin/testVector
CMakeFiles/testVector.dir/testVector.cpp.o: In function `Drake::getDrakePath()':
testVector.cpp:(.text+0x6b): undefined reference to `__cxa_guard_acquire'
testVector.cpp:(.text+0x90): undefined reference to `std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&)'
testVector.cpp:(.text+0x9a): undefined reference to `__cxa_guard_release'
testVector.cpp:(.text+0xa9): undefined reference to `std::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()'
testVector.cpp:(.text+0xc6): undefined reference to `__cxa_guard_abort'
```
4.7 and 4.9 are fine.

### PR DESCRIPTION
**Pull request by jamiesnape**
This closes #960, closes #1281, and closes #1649.

PTAL @RussTedrake.

### CODE REVIEW

**Comment by jamiesnape** [comment]
@RussTedrake drake006 is not using `gcc-4.9`. Double check that `CC=gcc-4.9` and `CXX=g++-4.9`.

**Comment by RussTedrake** [comment]
forgot to update the crontab. should be fixed now.

**Comment by jamiesnape** [comment]
@RussTedrake How do I force drake006 to rebuild?

**Comment by RussTedrake** [comment]
closing and reopening was the right thing.  it's just one machine, so puts jobs in a queue.  you're in the queue:
```
russt@drake006:~$ more ~/drake-admin/www/drake_test_queue.txt
--pull=1862 --repo=https://github.com/jamiesnape/drake.git --tag=gcc-4.9 -DWITH_
ALL_SUPPORTED_EXTERNALS=ON
--pull=1865 --repo=https://github.com/jhoare/drake.git --tag=bot_core_lcmtypes_s
hell_path -DWITH_ALL_SUPPORTED_EXTERNALS=ON
```

**Comment by jamiesnape** [comment]
Thanks.

**Comment by jamiesnape** [comment]
Strange.
```
-- The C compiler identification is GNU
-- The CXX compiler identification is GNU
-- Check for working C compiler: /usr/bin/gcc-4.9
-- Check for working C compiler: /usr/bin/gcc-4.9 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/g++-4.9
-- Check for working CXX compiler: /usr/bin/g++-4.9 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
CMake Error at CMakeLists.txt:5 (message):
  GCC version must be at least 4.9
```

**Comment by jamiesnape** [comment]
@RussTedrake Is drake006 an Ubuntu 12.04 machine?

**Comment by RussTedrake** [comment]
Yes

**Comment by jamiesnape** [comment]
We may have an issue with the CMake version. We are investigating now.

**Comment by jamiesnape** [comment]
@RussTedrake If I can find a suitable PPA for 12.04, would you consider increasing the minimum CMake version to 2.8.10 (the default for 14.04)?

**Comment by RussTedrake** [comment]
Yes.  Would also have to update the installation instructions.

In fact, we failed to update the instructions (and install prereqs scripts) for gcc 4.9.  Can you take that, too?

**Comment by jamiesnape** [comment]
The documentation update for GCC 4.9 should be in this PR. I will update as necessary for CMake and look at the install prereqs scripts.

**Comment by RussTedrake** [comment]
Ah yes.  Hadn't reviewed yet.  Very nice!

**Comment by jamiesnape** [comment]
@billhoffman has confirmed that we need 2.8.10. I will look for a PPA for 12.04. This will work with the default version installed on 14.04.

**Comment by jamiesnape** [comment]
@RussTedrake You mentioned a while back about consolidating all the install prereqs files into one at the root. Is that still something to which you are open? It will be just as quick for me to do that now, rather than fix them all recursively.

**Comment by RussTedrake** [comment]
Yes.  See the exchange on the (just closed) PR about requiring java

**Comment by jamiesnape** [comment]
Cool. I will fix that now.

**Comment by jamiesnape** [comment]
@RussTedrake Do you still support Cygwin?

**Comment by RussTedrake** [comment]
Just discussed this on a different thread.  The old build servers rely on it.  Yes, would be good to maintain.

**Comment by jamiesnape** [comment]
Ok.

**Comment by jamiesnape** [comment]
@RussTedrake Can you run the new `install_prereqs.sh` from this PR on drake006?

**Comment by RussTedrake** [comment]
done.

**Comment by jamiesnape** [comment]
@RussTedrake Looks like there is a problem with drake006.

**Comment by RussTedrake** [comment]
i tried running install_prereqs again. it seems ok

**Comment by RussTedrake** [comment]
@jamiesnape -- this PR is breaking the reporting script because the build server's attempt to run the tests did not result in a Test.xml being generated in the ctest directory.

ctest is being run with 
```
ctest -j $numcores -D Experimental -C Release --output-on-failure --timeout 1000 || true
```

**Comment by jamiesnape** [comment]
Can you email me the log files?

**Comment by jamiesnape** [comment]
On the small amount of output that I can see, it says "No such file or directory", but I do not know to which file/directory that refers.

**Comment by jamiesnape** [comment]
@RussTedrake I just ran `runUnitTests` on an EC2 instance provisioned with the scripts in this PR and the runs the tests fine. Are you running some other command on `drake006`?

**Comment by RussTedrake** [comment]
no. just runUnitTests. just looked in that directory… it looks like the build actually failed. trying to reproduce now.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+cpplint
```
Comment (RussTedrake) [inline requirements.txt]: can't you use a bash variable instead, and avoid this extra file?

File: drake/doc/fedora.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+*************************************************
+Centos / Red Hat Enterprise Linux 7.2 / Fedora 23
```
Comment (RussTedrake) [inline drake/doc/fedora.rst]: I think we should list this as (UNSUPPORTED) by annotating the title of this file.

File: requirements.txt (hunk 1 above)
Comment (jamiesnape) [inline requirements.txt]: This is best practice.

File: drake/doc/fedora.rst (hunk 2 above)
Comment (jamiesnape) [inline drake/doc/fedora.rst]: Sure.

File: drake/doc/homebrew.rst [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
-Now you can install the prerequistes::
+You can install the prerequisites using the ``install_prereqs.sh`` script::
 
 	brew update
-	brew install cmake
+	brew upgrade  # optional
 	cd drake-distro
-	make options # use the gui to choose which externals you want, then press c to configure, then g to generate makefiles and exit
-	make download-all
 	./install_prereqs.sh homebrew
```
Comment (RussTedrake) [inline drake/doc/homebrew.rst]: I think we should go all of the way with this PR and remove the install_prereqs files completely.  We should simply copy the working lines in install prereqs into the instructions here.   my understanding is that this is the standard workflow that people expect, and provides extra visibility.

Comment (jamiesnape) [inline drake/doc/homebrew.rst]: Sounds good.

**Comment by jamiesnape** [comment]
The install_prereqs are gone and documentation is updated. Just need to make sure that Travis CI passes.

**Comment by RussTedrake** [comment]
beautiful.  thank you.

**Comment by RussTedrake** [comment]
Note that this now requires a worst-case install for everyone... e.g. everyone has to install vtk5 even if they don't choose director.  i also worry that it will be hard to remove dependencies from this list (e.g. if we move to supporting some of the externals as precompiled instead of from source). but i've already expressed those reservations and agree this is the best/common practice.

developers -- last chance to comment.

**Comment by jamiesnape** [comment]
Travis CI and Jenkins are both happy.

**Pull request merged**